src/groth16/prover.rs

```diff
@@ -16,6 +16,22 @@ use crate::multicore::Worker;
 use crate::multiexp::{gpu_multiexp_supported, multiexp, DensityTracker, FullDensity};
 use crate::{Circuit, ConstraintSystem, Index, LinearCombination, SynthesisError, Variable};
 
+// We check to see if another higher priority process needs to use 
+// the GPU for each multiexp
+macro_rules! check_for_higher_prio {
+  () => { true }
+}
+
+#[cfg(feature = "gpu")]
+macro_rules! check_for_higher_prio {
+  () => {
+      match gpu::gpu_is_not_acquired() {
```

Comment:
FYI: this match clause could also be written as:

```rust
gpu::gpu_is_not_acquired().unwrap_or(false)
```

Which might make your code simpler.